tests/core/config/test_project_settings.h

```diff
@@ -97,6 +105,50 @@ TEST_CASE("[ProjectSettings] Set value should be returned when retrieved") {
 	CHECK(ProjectSettings::get_singleton()->has_setting("my_custom_setting"));
 }
 
+TEST_CASE("[ProjectSettings] localize_path") {
+	String old_resource_path = TestProjectSettingsInternalsAccessor::resource_path();
+	TestProjectSettingsInternalsAccessor::resource_path() = DirAccess::create(DirAccess::ACCESS_FILESYSTEM)->get_current_dir();
+	String root_path = ProjectSettings::get_singleton()->get_resource_path();
+	String root_path_win = ProjectSettings::get_singleton()->get_resource_path().replace("/", "\\");
+
+	CHECK_EQ(ProjectSettings::get_singleton()->localize_path("filename"), "res://filename");
+	CHECK_EQ(ProjectSettings::get_singleton()->localize_path("path/filename"), "res://path/filename");
+	CHECK_EQ(ProjectSettings::get_singleton()->localize_path("path\\filename"), "res://path/filename");
+	CHECK_EQ(ProjectSettings::get_singleton()->localize_path("path/something/../filename"), "res://path/filename");
+	CHECK_EQ(ProjectSettings::get_singleton()->localize_path("path\\something\\..\\filename"), "res://path/filename");
+	CHECK_EQ(ProjectSettings::get_singleton()->localize_path("path/./filename"), "res://path/filename");
+	CHECK_EQ(ProjectSettings::get_singleton()->localize_path("path\\.\\filename"), "res://path/filename");
+
+	// FIXME?: These checks pass, but that doesn't seems correct
```

Comment:
These have passed before as well, this behavior isn't new. Only the last 4 checks are actually affected by this PR.